integration_test.go

```diff
@@ -0,0 +1,539 @@
+	"localSymbolTableWithMultipleSymbolsAndImportsFields.ion",
+	"localSymbolTableWithMultipleSymbolsFields.10n",
+	"localSymbolTableWithMultipleSymbolsFields.ion",
+	"longStringRawControlCharacter.ion",
+	"minLongWithLenTooLarge.10n",
+	"minLongWithLenTooSmall.10n",
+	"month_1.ion",
+	"month_2.ion",
+	"negativeIntZero.10n",
+	"negativeIntZeroLn.10n",
+	"nopPadTooShort.10n",
+	"nopPadWithAnnotations.10n",
+	"nullDotCommentInt.ion",
+	"offsetHours_1.ion",
+	"offsetHours_2.ion",
+	"offsetMinutes_1.ion",
+	"offsetMinutes_2.ion",
+	"offsetMinutes_3.ion",
+	"sexpOperatorAnnotation.ion",
+	"stringLenTooLarge.10n",
+	"stringRawControlCharacter.ion",
+	"stringWithLatinEncoding.10n",
+	"structOrderedEmpty.10n",
+	"surrogate_1.ion",
+	"surrogate_10.ion",
+	"surrogate_2.ion",
+	"surrogate_4.ion",
+	"surrogate_5.ion",
+	"surrogate_6.ion",
+	"surrogate_7.ion",
+	"surrogate_8.ion",
+	"surrogate_9.ion",
+	"symbolIDUnmapped.10n",
+	"symbolIDUnmapped.ion",
+	"symbolLenTooLarge.10n",
+	"timestampHourWithoutMinute.10n",
+	"timestampLenTooLarge.10n",
+	"timestampSept31.10n",
+	"timestamp_0000-00-00.ion",
+	"timestamp_0000-00-00T.ion",
+	"timestamp_0000-00-01.ion",
+	"timestamp_0000-00-01T.ion",
+	"timestamp_0000-00T.ion",
+	"timestamp_0000-01-00.ion",
+	"timestamp_0000-01-00T.ion",
+	"timestamp_0000-01-01.ion",
+	"timestamp_0000-01-01T.ion",
+	"timestamp_0000-01T.ion",
+	"timestamp_0000-12-31.ion",
+	"timestamp_0000T.ion",
+	"timestamp_0001-00-00.ion",
+	"timestamp_0001-00-00T.ion",
+	"timestamp_0001-00-01.ion",
+	"timestamp_0001-00-01T.ion",
+	"timestamp_0001-00T.ion",
+	"timestamp_0001-01-00.ion",
+	"timestamp_0001-01-00T.ion",
+	"timestamp_10.ion",
+	"timestamp_11.ion",
+	"timestamp_5.ion",
+	"timestamp_6.ion",
+	"timestamp_7.ion",
+	"timestamp_8.ion",
+	"timestamp_9.ion",
+	"year_3.ion",
+}
+
+func TestBinaryRoundTrip(t *testing.T) {
+	readFilesAndTest(t, goodPath, binaryRoundTripSkipList, func(t *testing.T, path string) {
+		binaryRoundTrip(t, path)
+	})
+}
+
+func TestTextRoundTrip(t *testing.T) {
+	readFilesAndTest(t, goodPath, textRoundTripSkipList, func(t *testing.T, path string) {
+		textRoundTrip(t, path)
+	})
+}
+
+func TestLoadBad(t *testing.T) {
+	readFilesAndTest(t, badPath, malformedIonsSkipList, func(t *testing.T, path string) {
+		testLoadBad(t, path)
+	})
+}
+
+func binaryRoundTrip(t *testing.T, fp string) {
+	b := loadFile(t, fp)
+
+	// Make a binary writer from the file
+	r := NewReaderBytes(b)
+	buf := bytes.Buffer{}
+	bw := NewBinaryWriter(&buf)
+	writeToWriterFromReader(t, r, bw)
+	bw.Finish()
+
+	// Make a text writer from the binary writer
+	r = NewReaderBytes(buf.Bytes())
+	str := strings.Builder{}
+	tw := NewTextWriter(&str)
+	writeToWriterFromReader(t, r, tw)
+	tw.Finish()
+
+	// Make another binary writer using the text writer
+	r = NewReaderStr(str.String())
+	buf2 := bytes.Buffer{}
+	bw2 := NewBinaryWriter(&buf2)
+	writeToWriterFromReader(t, r, bw2)
+	bw2.Finish()
+
+	// Compare the 2 binary writers
+	if !reflect.DeepEqual(bw, bw2) {
```

Comment:
I think tihs is ok for now. Doing a little reading online it seems like reflect.DeepEqual has some caveats (or rather documented behaviour ) we should take into account when it comes to checking equality of Ion values per the ion spec. 
https://github.com/google/go-cmp

This seems to be a commonly (according to my google searches) Go lib that also allows customization through `Equal` interface.